File: 27--Jackpot-Redux/Jackpot Redux/Ticket.m
//
//  Ticket.m
//  Jackpot Redux
//
//  Created by Michael Reynolds on 11/10/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//

#import "Ticket.h"

@implementation Ticket

+ (Ticket *)newTicket
{
    Ticket *aTicket = [[Ticket alloc] init];
    NSMutableArray *numbers = [[NSMutableArray alloc] init];
    BOOL willLoop = true;
    while(willLoop)
    {
        [numbers removeAllObjects];
        while (numbers.count < 6)
        {
            int number = 1 + arc4random() % 52;
            NSNumber *pick = [NSNumber numberWithInt:number];
            [numbers addObject: pick];
        }
        
        NSSet *setFromArray = [NSSet setWithArray:numbers];
        if(setFromArray.count == numbers.count)
        {
            willLoop = false;
        }
    }
    aTicket.numbers = numbers;
    return aTicket;
}

- (BOOL)noDuplicates:(NSMutableArray *)arrayToTest
{
    NSSet *setFromArray = [NSSet setWithArray:arrayToTest];
    if(setFromArray.count == arrayToTest.count)
    {
        return true;
    }
    else
    {
        return false;
    }
}

- (NSString *)checkWinningTicket:(NSMutableArray *)winningTicketNumbers
{
    int matchingNumbers = 0;
    
    NSString *prize = [[NSString alloc] init];
    prize = @"$0";
    self.isWinner = NO;
    
    for (NSNumber *winningNumber in winningTicketNumbers)
    {
        for(NSNumber *number in self.numbers)
        {
            if([number isEqualToNumber: winningNumber])
            {
                matchingNumbers += 1;
            }
        }
    }
    
//    NSLog(@"%@", self.numbers);
//    NSLog(@"%@", winningTicketNumbers);
    
    if (matchingNumbers == 3)
    {
        prize = @"$1";
        self.isWinner = YES;
    }
    else if (matchingNumbers == 4)
    {
        prize = @"$5";
        self.isWinner = YES;
    }
    else if (matchingNumbers == 5)
    {
        prize = @"$20";
        self.isWinner = YES;
    }
    else if (matchingNumbers == 6)
    {
        prize = @"$100";
        self.isWinner = YES;        
    }
    
    NSLog(@"tick prz amt %d", matchingNumbers);
    return prize;
}


@end

File: 07--Jackpot/jackpot/jackpot/Ticket.swift

//
//  Ticket.swift
//  jackpot
//
//  Created by david on 10/13/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//


import Foundation

class Ticket
{
    var ticketString = ""
    var ticketArray = Array<Int>()
    var winningTicket = Array<Int>()

    
    init()
    {
        ticketArray = makeTicket()
        ticketString = formatTicket(ticketArray)
    }
    
    init(arrayFromPicker: Array<Int>)
    {
        winningTicket = arrayFromPicker
    }

    func makeTicket() -> Array<Int>
    {
        var willLoop = true
        while willLoop == true
        {
            ticketArray = []
            
            while ticketArray.count < 6
            {
                let number = Int(1 + arc4random() % 52)
                ticketArray.append(number)
                
            }
            
            if checkDuplicateNumbers(ticketArray) == true
            {
                willLoop = false
                return ticketArray
            }
        }
    }
    
    func checkDuplicateNumbers(arrayToTest: Array<Int>) -> Bool
    {
        let arrayFromSet = Set<Int>(arrayToTest)
        if arrayToTest.count == arrayFromSet.count
        {
            return true
        }
        else
        {
            return false
        }
    }

    func formatTicket(ticketArray: Array<Int>) -> String
    {
        var ticketAsString = ""
        for number in ticketArray
        {
            ticketAsString = ticketAsString + "\(number)" + " "
        }
        return ticketAsString
    }
    
    
    func checkWinningTicket(winningTicketNum: Array<Int>) -> String
    {
        var matchingNumbers = 0
        var prizeAmount = 0
        for x in winningTicketNum
        {
            if ticketArray.contains(x)
            {
                matchingNumbers += 1
            }
        }
        if matchingNumbers == 3
        {
            prizeAmount = 1
        }
        if matchingNumbers == 4
        {
            prizeAmount = 5
        }
        if matchingNumbers == 5
        {
            prizeAmount = 20
        }
        if matchingNumbers == 6
        {
            prizeAmount = 100
        }
        
        return "$ " + String(prizeAmount)
        
    }
}
